src/methods/clipping/clipping_processor.jl

```diff
@@ -1,6 +1,59 @@
 # # Polygon clipping helpers
 # This file contains the shared helper functions for the polygon clipping functionalities.
 
+# This file specifically defines helpers for the Foster-Hormann clipping algorithm.
+
+
+"""
+    abstract type IntersectionAccelerator
+
+The abstract supertype for all intersection accelerator types.
+
+The idea is that these speed up the edge-edge intersection checking process,
+perhaps at the cost of memory.
+
+The naive case is [`NestedLoop`](@ref), which is just a nested loop, running in O(n*m) time.
+
+Then we have [`SingleSTRtree`](@ref), which is a single STRtree, running in O(n*log(m)) time.
+
+Then we have [`DoubleSTRtree`](@ref), which is am simultaneous double-tree traversal of two STRtrees.
+
+Finally, we have [`AutoAccelerator`](@ref), which is an automatic accelerator that chooses the best
+accelerator based on the size of the input polygons.  This gets materialized in build_a_list for now.
+"""
+abstract type IntersectionAccelerator end
+struct NestedLoop <: IntersectionAccelerator end
+struct SingleSTRtree <: IntersectionAccelerator end
+struct DoubleSTRtree <: IntersectionAccelerator end
+struct SingleNaturalTree <: IntersectionAccelerator end
+struct DoubleNaturalTree <: IntersectionAccelerator end
+struct ThinnedDoubleNaturalTree <: IntersectionAccelerator end
```

Comment:
Is this the only one with thinning?

Wondering if it can be a type parameter

Reply:
No, we could construct a thinned tree as well

In all honesty the thinning ones will only get used when you're constructing a tree from scratch.  Otherwise (in the ideal world) AutoAccelerator will pick up on it and force that particular tree / pattern to be used unless it's really suboptimal.